From ce3ddc6e7575da719ba0cc92092b599ad47fe490 Mon Sep 17 00:00:00 2001
From: Aethersailor <22260104+Aethersailor@users.noreply.github.com>
Date: Sun, 14 Dec 2025 02:16:57 +0800
Subject: [PATCH] chore: hide subconverter fingerprints when fetching

---
 src/handler/webget.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/handler/webget.cpp b/src/handler/webget.cpp
index 4433536..e0a4b52 100644
--- a/src/handler/webget.cpp
+++ b/src/handler/webget.cpp
@@ -31,7 +31,7 @@ std::mutex cache_rw_lock;
 RWLock cache_rw_lock;
 
 //std::string user_agent_str = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36";
-static auto user_agent_str = "subconverter/" VERSION " cURL/" LIBCURL_VERSION;
+static auto user_agent_str = "clash.meta";
 
 struct curl_progress_data
 {
@@ -178,8 +178,8 @@ static int curlGet(const FetchArgument &argument, FetchResult &result)
         if(!argument.request_headers->contains("User-Agent"))
             curl_easy_setopt(curl_handle, CURLOPT_USERAGENT, user_agent_str);
     }
-    header_list = curl_slist_append(header_list, "SubConverter-Request: 1");
-    header_list = curl_slist_append(header_list, "SubConverter-Version: " VERSION);
+    else
+        curl_easy_setopt(curl_handle, CURLOPT_USERAGENT, user_agent_str);
     if(header_list)
         curl_easy_setopt(curl_handle, CURLOPT_HTTPHEADER, header_list);
 
-- 
2.52.0.windows.1

